certora/specs/Splits.spec

```diff
@@ -0,0 +1,418 @@
+//// ## Verification of Splits.sol
+//// 
+//// `Splits.sol` is part of the Drips protocol
+//// 
+//// The splits functionality, as explained in DripsHub.sol by the developers:
+//// The user can share collected funds with other users by using splits.
+//// When collecting, the user gives each of their splits receivers a fraction of the received funds.
+//// Funds received from splits are available for collection immediately regardless of the cycle.
+//// They aren't exempt from being split, so they too can be split when collected.
+//// Users can build chains and networks of splits between each other.
+//// Anybody can request collection of funds for any user,
+//// which can be used to enforce the flow of funds in the network of splits.
+//// 
+//// ### Assumptions and Simplifications
+//// - All the internal methods are wrapped in CVT callable versions
+//// - Dummy functions used to control the SplitsReceiver[] receivers
+//// - When testing the split weight functionality we safely assumed weight < _TOTAL_SPLITS_WEIGHT
+////   since the method _assertSplitsValid() validates them before assignment
+//// 
+//// ### Properties
+
+
+/// @notice Calculate results of splitting an amount using the current splits configuration.
+/// We verify that amount == collectableAmt + splitAmt, including that no overflow occurs
+rule correctnessOfSplitResults() {
```

Comment:
Since you're compiling with Solidity >0.8.0 then there are no overflows.
In that case, the third assert is sufficient.

Reply:
You are right, I will remove the obsolete code.